File: binarysearch.cpp
 #include<algorithm>
#include<iostream>
using namespace std;
int main()
{
	int n,i,j,k,ele,l,h,mid,a[50],count,fpos,lpos;
	cout<<"Enter the number of elements in array and element to be searched\n";
	cin>>n>>ele;
	l = 0;
	h = n-1;
	cout<<"Enter the elements of array\n";
	for(j=0;j<n;j++)
		cin>>a[j];
	sort(a, a+n);
	while(l<=h)
	{
		mid = (int)(l+h)/2;
		if(a[mid]==ele)
		{	
			i = mid;
			k = mid;			
			while(a[i]==ele)
			{			
				fpos = i;
				i--;
			}
			while(a[k]==ele)
			{
				lpos = k;
				k++;
			}
			cout<<"The first occurrence of the key is at "<<fpos<<" and last occurrence is at "<<lpos<<"\n";
			count = lpos-fpos+1;
			cout<<"The count is "<<count<<"\n";		
			break;
		}
		else if(a[mid]<ele)
			l = mid+1;
		else 
			h = mid-1;
	}
	if(l>h)
		cout<<"-1 (Element is not present)\n";
	return 0;
}

File: binarysearch.py
def binary(L,x):
    L.sort()
    l = 0
    r = len(L)-1
    while l <= r:
        mid = int((l+r)/2)
        if L[mid] == x:
            i = mid
            k = mid
            while L[i] == x and i != 0:
                f_pos = i
                i -= 1
            while L[k] == x and k != r:
                l_pos = k
                k += 1
            print('The first occurrence of the key is at:',f_pos,'\nLast occurrence is at ',l_pos)
            print('Count:',(l_pos-f_pos+1))
            return
        elif L[mid] < x:
            l = mid+1
        else :
            r = mid-1
        
    else:
        print('Element not found.')
        return
    

L = list(map(int, input('Enter the Array:').split()))
binary(L, int(input('Enter the search element:')))
